### PR TITLE
fix: what fluid to recirculate when there is no rate entering a compressor stage in a multiple streams compressor train at a given time step

### PR DESCRIPTION
**Pull request by olelod**
This regards VairableSpeedCompressorTrainCommonShaftMultipleStreamsAndPressures only. In situations where there is both an additional ingoing stream and an additional ingoing stream between two compressors, one can end up with a situation where one tries to mix together two streams with zero mass rate (to mix the one stream passing from compressor 1 to compressor 2 and the additional stream entering the compressor train between compressor 1 and compressor 2). This means that if we look at this time step isolated there is an unknown composition to recirculate in the second compressor. This fluid to recirculate should have the same composition as the fluid passing through the second compressor at the previous time step.

## What does this pull request change?

- Includes a class attribute keeping track of the composition of the fluid passing through the different compressor stages in a compressor train, such that it will be available for the next time step (except for the first time step - there the code will fail) 
- A small fix giving the correct standard condition density for the resulting fluid with mixing two fluids (now using the correct pressure and temperature, instead of assuming that the given pressure and temperature is the standard condition pressure and standard condition temperature)
- Adding a test

### CODE REVIEW

File: src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py [hunk 1]
```diff
@@ -277,7 +278,5 @@ def mix_in_stream(
 
         return FluidStream(
             existing_fluid=mixed_neqsim_fluid_stream,
-            temperature_kelvin=temperature_kelvin,
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py:280]: so, this was a bug?

Comment (TeeeJay) [reply src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py:280]: or just unnecessary?

File: src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py [hunk 2]
```diff
@@ -879,7 +892,7 @@ def calculate_compressor_train_given_rate_ps_speed(
             )
 
             mass_rate_previous_stage_kg_per_hour = mass_rate_this_stage_kg_per_hour
-
+            self.fluid_to_recirculate_in_stage[stage_number] = inlet_stream
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:895]: possibly to replace set and get to this member variable using methods to more clearly indicate that we are adding "leftover rate" in one timestep that is "captured" in next timestep? e.g.

`set_recirculate_fluid_in_next_timestep(stage, stream)`

and

`get_fluid_to_recirculate(stage)`

or some other names that make more sense? Does it make sense?

Comment (TeeeJay) [reply src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:895]: But is it actually "next timestep"? Isnt it "next recirculation round" or whatever name is being used by experts? to not be confused with "ecalc timesteps"?

Comment (TeeeJay) [reply src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:895]: unless, I am wrong ;)

File: src/ecalc/libraries/libecalc/common/tests/core/models/compressor_modelling/test_variable_speed_compressor_train_multiple_streams.py [hunk 3]
```diff
@@ -219,6 +219,41 @@ def variable_speed_compressor_train_two_compressors_two_streams(
     )
 
 
+@pytest.fixture
```
Comment (TeeeJay) [inline src/ecalc/libraries/libecalc/common/tests/core/models/compressor_modelling/test_variable_speed_compressor_train_multiple_streams.py:222]: :+1:

Comment (TeeeJay) [reply src/ecalc/libraries/libecalc/common/tests/core/models/compressor_modelling/test_variable_speed_compressor_train_multiple_streams.py:222]: great with test. and it seems very minimal. Win! :)

We should have more tests like these, for all the edge cases we can think of. Just a reminder for future and improvement of tests coverage :)

File: src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py (hunk 1 above)
Comment (olelod) [reply src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py:280]: Just unnecessary. When existing_fluid is defined, the pressure and temperature is taken from there. The bug was line 61 and 62. pressure_bara and temperature_kelvin was always defaulted to UnitConstants.STANDARD_PRESSURE_BARA and UnitConstants.STANDARD_TEMPERATURE_KELVIN, so the if statement would always evaluate to true...

File: src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py (hunk 2 above)
Comment (olelod) [reply src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py:895]: > possibly to replace set and get to this member variable using methods to more clearly indicate that we are adding "leftover rate" in one timestep that is "captured" in next timestep? e.g.
> 
> `set_recirculate_fluid_in_next_timestep(stage, stream)`
> 
> and
> 
> `get_fluid_to_recirculate(stage)`
> 
> or some other names that make more sense? Does it make sense?

Yes, surely. Was a bit unsure how to approach it, but that sounds better.

**Review by TeeeJay — APPROVED** [review]

**Review by frodehk — APPROVED** [review]
